src/Libraries/CoreNodeModels/DropDown.cs

```diff
@@ -89,6 +89,27 @@ public int SelectedIndex
             }
         }
 
+        private string selectedString = "";
+        public string SelectedString
+        {
+            get { return selectedString; }
+            set
+            {
+                if(string.IsNullOrEmpty(selectedString))
+                {
+                    if(Items.Count()>0 && !string.IsNullOrEmpty(value))
+                    {
+                        var item = Items.FirstOrDefault(i => (i.Item).ToString().Equals(value));
+                        SelectedIndex = item != null ?
```

Comment:
This reset SelectedIndex action is not quite fit the set method's function, but I can't find any other place to do this action. Do you have any other suggestions?

Reply:
Hi @QilongTang , I have debugged in dropdown class, and there is a JsonConstructor now at https://github.com/DynamoDS/Dynamo/blob/0b0d90ba1bf6cad239b7f44522d981981fbd5511/src/Libraries/CoreNodeModels/DropDown.cs#L99 , and Dynamo DropDown Items would be set here. And then, SelectedIndex and SelectedString was set value later. If I want to get the matched SelectedIndex and SelectedString, the values of Index, String and Items is necessary.